### PR TITLE
NodePort multi-dev support and BPF masq* tests

### LINKED ISSUE #9620: Support exposing services through multiple devices
NodePort services are currently exposed through the native device which has the default route on the host or a user specified device. In tunneling mode, they are additionally exposed through the tunnel interface (`cilium_vxlan` or `cilium_geneve`). This issue tracks exposing services through multiple native devices.

### PR DESCRIPTION
**Pull request by brb**
This PR adds a multi-dev support for NodePort BPF and BPF MASQ.

The initial review was done in #10878. Reviewable per commit.

The follow-up tasks https://github.com/cilium/cilium/issues/11018.

Fix #9620.

```release-note
Allow attaching BPF NodePort and BPF masquerade to multiple devices
```

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

File: common/utils.go [hunk 1]
```diff
@@ -50,6 +50,51 @@ func C2GoArray(str string) []byte {
 	return ret
 }
 
+// GoArray2C transforms a byte slice into its hexadecimal string representation.
```
Comment (aanm) [inline common/utils.go]: We need to move away of this package cc @joestringer

File: pkg/option/config.go [hunk 2]
```diff
@@ -1205,7 +1205,7 @@ type DaemonConfig struct {
 	LibDir           string     // Cilium library files directory
 	RunDir           string     // Cilium runtime directory
 	NAT46Prefix      *net.IPNet // NAT46 IPv6 Prefix
-	Device           string     // Receive device
+	Devices          []string   // bpf_netdev device
```
Comment (aanm) [inline pkg/option/config.go:1209]: Changing from 1 device to multiple devices requires a change in the flag name. please bring this up in the next weekly meeting

**Review by aanm — APPROVED** [review]

File: common/utils.go (hunk 1 above)
Comment (joestringer) [reply common/utils.go]: Filed https://github.com/cilium/cilium/issues/11293

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-focus K8sDatapath*

**Comment by brb** [comment]
CI is failing because `test-k8s2` cannot be started (it was working before rebasing against the deployment manager):
```
Warning  FailedScheduling  17s (x16 over 4m18s)  default-scheduler  0/3 nodes are available: 1 node(s) didn't have free ports for the requested pod ports, 1 node(s) didn't match node selector, 1 node(s) were not ready.
```

I'm currently investigating this.

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-focus K8sDatapath*

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-focus K8sService*

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by borkmann** [comment]
> Yep:
> 
> ```
> 63: foobar,;=a: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
>     link/ether ea:f2:b3:b8:78:79 brd ff:ff:ff:ff:ff:ff
> ```

Wow, good to know, sigh.

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-focus K8sDatapathConfig*

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-focus K8sDatapathConfig*

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-focus K8sDatapathConfig*

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-focus K8sDatapathConfig*

**Review by borkmann — APPROVED** [review]

**Review by rolinh — APPROVED** [review]

**Pull request merged**